can-route-pushstate.js

```diff
@@ -159,23 +161,31 @@ canReflect.assign(PushstateObservable.prototype, {
 
 			// If link is within the same domain and descendant of `root`.
 			if (window.location.host === linksHost) {
-				var root = cleanRoot();
+				var root = cleanRoot(),
+					pathname,
+					href,
+					nodePathWithSearch;
+
+				if (node instanceof HTMLAnchorElement) {
+					pathname = node.pathname;
+					href = node.href;
+					nodePathWithSearch = pathname + node.search;
+				} else if (node.namespaceURI === "http://www.w3.org/1999/xlink") {
+					pathname = href = node.getAttributeNS("http://www.w3.org/1999/xlink", "href");
+					nodePathWithSearch = href;
+				}
 
 				// If the link is within the `root`.
-				if (node.pathname.indexOf(root) === 0) {
-
-					// Removes root from url.
-					var nodePathWithSearch = node.pathname + node.search,
-						url = nodePathWithSearch.substr(root.length);
+				if (pathname.indexOf(root) === 0) {
```

Comment:
I think this should check for `pathname !== undefined` first in case the two cases above aren’t hit.

Reply:
Done!